### PR TITLE
Notify table owners

### PR DESCRIPTION
**Pull request by csteez**
### Summary of Changes

Sends an email to table owners, if any, when a JIRA ticket has been successfully created. 

### Tests

Modified tests to handle the new structure for the API calls. 

### Documentation

N/A

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes, including screenshots of any UI changes. 
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/static/js/ducks/issue/sagas.ts [hunk 1]
```diff
@@ -27,8 +27,20 @@ export function* getIssuesWatcher(): SagaIterator {
 export function* createIssueWorker(action: CreateIssueRequest): SagaIterator {
   try { 
     let response;
-    response = yield call(API.createIssue, action.payload.data);
+    const { key, title, description, owners, resourceName, resourcePath, sender} = action.payload;
+    response = yield call(API.createIssue, key, title, description);
     yield put((createIssueSuccess(response)));
+
+    // Send a notification to table owners if they exist
```
Comment (csteez) [inline amundsen_application/static/js/ducks/issue/sagas.ts]: Curious what people think about this as a way to send an email- I felt like it made the most sense to trigger the event here. Another question I have is whether other people think the notification for sending an email should be hidden? Or maybe there should be some text on the modal that the table owners will be notified?

File: amundsen_application/api/utils/notification_utils.py [hunk 2]
```diff
@@ -122,6 +129,11 @@ def get_notification_html(*, notification_type: str, options: Dict, sender: str)
             comment = ('<br/>{sender} has included the following information with their request:'
                        '<br/>{comment}<br/>').format(sender=sender, comment=options_comment)
 
+    if notification_type == NotificationType.DATA_ISSUE_CREATED:
+        greeting = 'Hello data owner,<br>'
```
[the remark this answers is not in the document]
Comment (csteez) [reply amundsen_application/api/utils/notification_utils.py]: I altered this per Mark's suggestion - I don't feel strongly on it and I'm happy to get rid of the extra text.

File: amundsen_application/proxy/issue_tracker_clients/jira_client.py [hunk 3]
```diff
@@ -67,7 +67,6 @@ def create_issue(self, table_uri: str, title: str, description: str) -> DataIssu
         :param description: Description of the Jira issue
         :param table_uri: Table Uri ie databasetype://database/table
         :param title: Title of the Jira ticket
-        :return: Metadata about the newly created issue
```
[the remark this answers is not in the document]
Comment (csteez) [reply amundsen_application/proxy/issue_tracker_clients/jira_client.py]: Probably accidental, good catch!

File: amundsen_application/static/js/ducks/issue/sagas.ts [hunk 4]
```diff
@@ -27,8 +27,20 @@ export function* getIssuesWatcher(): SagaIterator {
...
+    const { key, title, description, owners, resourceName, resourcePath, sender} = action.payload;
+    response = yield call(API.createIssue, key, title, description);
     yield put((createIssueSuccess(response)));
+
+    // Send a notification to table owners if they exist 
+    if (owners && owners.length > 0) {
+      yield put(submitNotification(owners, sender, NotificationType.DATA_ISSUE_CREATED, {
+        resource_name: resourceName, 
+        resource_path: resourcePath, 
+        description_requested: false,
```
[the remark this answers is not in the document]
Comment (csteez) [reply amundsen_application/static/js/ducks/issue/sagas.ts]: great point!

File: amundsen_application/static/js/components/TableDetail/ReportTableIssue/index.tsx [hunk 5]
```diff
@@ -10,18 +10,30 @@ import './styles.scss';
...
+import { TableMetadata } from 'interfaces';
 
 export interface ComponentProps {
   tableKey: string;
   tableName: string;
 }
 
 export interface DispatchFromProps {
-  createIssue: (data: FormData) => CreateIssueRequest; 
+  createIssue: (
```
[the remark this answers is not in the document]
Comment (csteez) [reply amundsen_application/static/js/components/TableDetail/ReportTableIssue/index.tsx]: great suggestion - definitely doing that.

File: amundsen_application/static/js/components/TableDetail/ReportTableIssue/index.tsx [hunk 6]
```diff
@@ -101,8 +125,18 @@ export class ReportTableIssue extends React.Component<ReportTableIssueProps, Rep
   }
 }
 export const mapStateToProps = (state: GlobalState) => {
+  const ownerObj = state.tableMetadata.tableOwners.owners; 
+  const tableOwnersEmails = []; 
+  Object.keys(ownerObj).map((ownerId) => {
```
[the remark this answers is not in the document]
Comment (csteez) [reply amundsen_application/static/js/components/TableDetail/ReportTableIssue/index.tsx]: oh good point, thanks!

File: amundsen_application/static/js/ducks/issue/api/v0.ts [hunk 7]
```diff
@@ -22,11 +24,20 @@ export function getIssues(tableKey: string) {
...
-  const headers =  {'Content-Type': 'multipart/form-data' };
-  return axios.post(`${API_PATH}/issue`, data, { headers }
-    ).then((response: AxiosResponse<IssueApi>) => {
+export function createIssue(payload: CreateIssuePayload, notificationPayload: NotificationPayload) {
+  return axios.post(`${API_PATH}/issue`, {
+    key: payload.key, 
+    title: payload.title, 
+    description: payload.description
+  })
+  .then((response: AxiosResponse<IssueApi>) => {
```
[the remark this answers is not in the document]
Comment (ttannis) [reply amundsen_application/static/js/ducks/issue/api/v0.ts]: I also do want to call out that as is, if issue created succeeds but the notification fails then the user will not see their issue created. Our owner editing flow suffers from this as well *but* in that case we made it such that if either options fails, the UI tells the user to refresh the page. When the page is refreshed the user will be able to see if their owner update took effect or not. So as a final improvement for this feature, can you file a ticket for addressing the failure case of issue creation in the UI? I'd consider it low priority though and it always be a starter task for someone else .

File: amundsen_application/api/utils/notification_utils.py [hunk 8]
```diff
@@ -20,6 +20,7 @@ class NotificationType(str, Enum):
     OWNER_REMOVED = 'owner_removed'
     METADATA_EDITED = 'metadata_edited'
     METADATA_REQUESTED = 'metadata_requested'
+    DATA_ISSUE_CREATED = 'data_issue_reported'
```
[the remark this answers is not in the document]
Comment (csteez) [reply amundsen_application/api/utils/notification_utils.py]: oh very good catch, thanks!

File: amundsen_application/static/js/components/TableDetail/ReportTableIssue/tests/index.spec.tsx [hunk 9]
```diff
@@ -12,8 +12,36 @@ import {
...
+  notificationType: 'data_issue_reported',
+  options: {
+    resource_name: 'schema.table_name',
+    resource_path: '/table_detail/cluster/database/schema/table_name',
+  },
+  recipients: ['owner@email'],
+  sender: 'user@email'
+}
+  
 // @ts-ignore: How to mock FormData without TypeScript error?
```
[the remark this answers is not in the document]
Comment (csteez) [reply amundsen_application/static/js/components/TableDetail/ReportTableIssue/tests/index.spec.tsx]: yeah, it still raises an error about FormData if I remove the `@ts-ignore` statement.

File: amundsen_application/static/js/ducks/issue/tests/index.spec.ts [hunk 10]
```diff
@@ -227,6 +280,24 @@ describe('issue ducks', () => {
...
+      it('creates notifications', () => {
+        return expectSaga(createIssueWorker, action)
+          .provide([
+            [matchers.call.fn(API.createIssue), issue],
+          ])
+          .put(createIssueSuccess(issue))
+          .run();
+      });
+
+      it('creates notifications', () => {
```
[the remark this answers is not in the document]
Comment (csteez) [reply amundsen_application/static/js/ducks/issue/tests/index.spec.ts]: yeah leftover from the test that tested notification behavior. i removed both of them as they aren't useful anymore.

File: amundsen_application/static/js/ducks/issue/api/v0.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (csteez) [reply amundsen_application/static/js/ducks/issue/api/v0.ts]: yeah very good point - will file a ticket as well

File: amundsen_application/static/js/components/TableDetail/ReportTableIssue/tests/index.spec.tsx [hunk 11]
```diff
@@ -12,8 +12,36 @@ import {
...
+ };
+
+const mockCreateIssuePayload = {
+  key: 'key', 
+  title: 'title', 
+  description: 'description'
+}
+
+const mockNotificationPayload = {
+  notificationType: 'data_issue_reported',
```
[the remark this answers is not in the document]
Comment (csteez) [reply amundsen_application/static/js/components/TableDetail/ReportTableIssue/tests/index.spec.tsx]: good catch!

**Pull request merged**